spec/models/membership_spec.rb

```diff
@@ -44,4 +44,16 @@
     membership = build(:membership, user:, group:)
     expect { membership.save(validate: false) }.to raise_error(ActiveRecord::RecordNotUnique)
   end
+
+  it "removes mismatched memberships" do
```

Comment:
Can we have a test for `destroy_invalid_organisation_memberships` that checks it doesn't affect membership of other users?